### PR TITLE
Creative: Extend search function to descriptions

### PR DESCRIPTION
**Pull request by srifqi**
It will also searches in node description, not just node name. Confused? Try search "nyan cat".

### CODE REVIEW

File: mods/creative/init.lua [hunk 1]
```diff
@@ -65,7 +65,10 @@ function creative_inventory.update(player_name, filter, tab_id)
 	for name, def in pairs(tab_category(tab_id)) do
 		if not (def.groups.not_in_creative_inventory == 1) and
 				def.description and def.description ~= "" and
-				(not filter or def.name:find(filter, 1, true)) then
+				(not filter or
```
Comment (kilbith) [inline mods/creative/init.lua]: Don't break the line here.

File: mods/creative/init.lua [hunk 2]
```diff
@@ -65,7 +65,10 @@ function creative_inventory.update(player_name, filter, tab_id)
...
+					def.name:find(filter, 1, true) or
+					def.description:lower():find(filter, 1, true)
```
Comment (kilbith) [inline mods/creative/init.lua]: You don't need to use `lower() here`, as the string is already lowered through the fields handler : https://github.com/minetest/minetest_game/blob/master/mods/creative/init.lua#L189

**Comment by kilbith** [comment]
Also I find the title of this PR inappropriate. Use _"Creative: extend search function to descriptions"_ instead.

File: mods/creative/init.lua [hunk 3]
```diff
@@ -65,7 +65,10 @@ function creative_inventory.update(player_name, filter, tab_id)
...
+				) then
```
Comment (kilbith) [inline mods/creative/init.lua]: ... and move this above.

File: mods/creative/init.lua (hunk 1 above)
Comment (srifqi) [inline mods/creative/init.lua]: How should I break the lines?

File: mods/creative/init.lua (hunk 2 above)
Comment (srifqi) [inline mods/creative/init.lua]: I think you missed a point. It lowers the filter, not the description.

Comment (kilbith) [inline mods/creative/init.lua]: Oops, my bad -- badly awaken.

File: mods/creative/init.lua (hunk 1 above)
Comment (srifqi) [inline mods/creative/init.lua]: Any guidelines to break lines?

Comment (kilbith) [inline mods/creative/init.lua]: Like so : 

``` Lua
(not filter or def.name:find(filter, 1, true) or
         def.description:lower():find(filter, 1, true)) then
```

Your line is short enough to do not be broken.

Comment (srifqi) [inline mods/creative/init.lua]: Done. 62173283e9f7d1e422a7bdc7673a97f3e7cd3ae8

**Comment by kilbith** [comment]
:+1: If commits squashed and commit name changed to the same as the PR title.

**Comment by srifqi** [comment]
Done. 50fcc7a7d00edb7bce02a2d8479bd2c0afb48536

**Comment by kaeza** [comment]
:+1:

**Comment by paramat** [comment]
:+1:

**Comment by Xanthin** [comment]
Thank you! It solves searching translated descriptions while the node names are still in english.

**Comment by Ekdohibs** [comment]
:+1:

**Comment by paramat** [comment]
Will merge soon.

**Comment by paramat** [comment]
9e54b379c8ba50228f3773a158cc5e4aa8b427a8

**Comment by srifqi** [comment]
Thanks!

**Pull request closed without merging**